### ISSUE TITLE
CA1508 False positive with char equality

### ISSUE THREAD

**Issue by meziantou** [dotnet/roslyn-analyzers#4387]
### Analyzer

**Diagnostic ID**: [CA1508](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1508): `Avoid dead conditional code`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

**Version**: v3.3.1 (Latest)

### Steps To Reproduce

````csharp
        private static bool TryParseUnit(string unit)
        {
            char last = unit[^1];
            if (last != 'b') // warning CA1508: 'last != 'b'' is always 'true'. Remove or refactor the condition(s) to avoid dead code.
                return false;

            return true;
        }
````

### Expected behavior

No diagnostic as the condition is not a constant.

### Actual behavior

Report a diagnostic

### Additional data

No diagnostic is reported if I inline the variable

````
if (unit[^1] != 'b') // no diagnostic
````

**Comment by mavasani** [comment]
Seems like the new IOperation for range/index needs to be handled in flow analysis.
